documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1189 @@
+    static abstract TVector BitDecrement(TVector x);
+    static abstract TVector BitIncrement(TVector x);
+    static abstract TVector FusedMultiplyAdd(TVector left, TVector right, TVector addend);
+    static virtual TVector FusedMultiplyAdd(TVector left, TScalar right, TVector addend) => TVector.FusedMultiplyAdd(left, TVector.Create(right), addend); 
+    static virtual TVector FusedMultiplyAdd(TVector left, TVector right, TScalar addend) => TVector.FusedMultiplyAdd(left, right, TVector.Create(addend)); 
+    static virtual TVector FusedMultiplyAdd(TVector left, TScalar right, TScalar addend) => TVector.FusedMultiplyAdd(left, TVector.Create(right), TVector.Create(addend));
+    static abstract TVector ReciprocalEstimate(TVector x);
+    static abstract TVector ReciprocalSqrtEstimate(TVector x);
+
+    // IFloatingPointIeee754<TVector>
+    static virtual TNewVector ILogB<TNewVector>(TVector x) where TNewVector : IVector<TNewVector, int> => TVector.ILogB<TNewVector, int>(x);
+    static abstract TNewVector ILogB<TNewVector, TInt>(TVector x) where TNewVector : IVector<TNewVector, TInt> where TInt : IBinaryInteger<TInt>;
+    static abstract TVector ScaleB(TVector x, Vector2D<int> n);
+    static abstract TVector ScaleB(TVector x, int n);
+    static virtual TNewVector RoundToInt<TNewVector>(TVector vector) where TNewVector : IVector<TNewVector, int> => TVector.RoundToInt<TNewVector, int>(vector);
+    static virtual TNewVector FloorToInt<TNewVector>(TVector vector) where TNewVector : IVector<TNewVector, int> => TVector.RoundToInt<TNewVector, int>(vector);
+    static virtual TNewVector CeilingToInt<TNewVector>(TVector vector) where TNewVector : IVector<TNewVector, int> => TVector.RoundToInt<TNewVector, int>(vector);
+    static abstract TNewVector RoundToInt<TNewVector, TInt>(TVector vector)
+        where TNewVector : IVector<TNewVector, TInt>
+        where TInt : IBinaryInteger<TInt>;
+    static abstract TNewVector FloorToInt<TNewVector, TInt>(TVector vector)
+        where TNewVector : IVector<TNewVector, TInt>
+        where TInt : IBinaryInteger<TInt>;
+    static abstract TNewVector CeilingToInt<TNewVector, TInt>(TVector vector)
+        where TNewVector : IVector<TNewVector, TInt>
+        where TInt : IBinaryInteger<TInt>;
+
+    static virtual Vector64<TScalar> AsVector64(TVector self) => Vector64.Create(TVector.AsSpan(self));
+    static virtual Vector128<TScalar> AsVector128(TVector self) => Vector128.Create(TVector.AsSpan(self));
+    static virtual Vector256<TScalar> AsVector256(TVector self) => Vector256.Create(TVector.AsSpan(self));
+    static virtual Vector512<TScalar> AsVector512(TVector self) => Vector512.Create(TVector.AsSpan(self));
+}
+
+public interface IVector2<TVector, TScalar> : IVector<TVector, TScalar>
+    where TVector : IVector2<TVector, TScalar>
+    where TScalar : INumberBase<TScalar>
+{
+    static virtual TVector UnitX => { get; }
+    static virtual TVector UnitY => { get; }
+    
+    TScalar X { get; }
```

Comment:
`X` could be on `IVector`, since it wouldn't make sense to define a 0'd dimension vector >.>